### PR TITLE
Bug 303

### PR DESCRIPTION
**Pull request by gilbertohasnofb**
@onox I just pushed my last commit trying to fix what we discuss at https://github.com/Juanvvc/c172p-detailed/commit/5112b2fdd83be4485a56c3b65178b011b2d493f8 . What do you think if we already test (and hopefully merge) this PR for the pitot tube and for the chokes? Then I will move on to create the wing straps (should be much smoother than it was today, as it will use the same parking brake logic we already implemented).

Also, I will open new issues to remind us to add these items to the checklist and tutorials later.

Note that I also set the pitot tube cover and chokes are set to false by default, so the first time someone opens the sim they will be able to fly without a problem (and will do the same for the wing straps). For those of us who want realism, then we can start setting them and their state is saved.

Let me know what you think, and thanks a lot for all the coaching, I really appreciate it :smile:

### CODE REVIEW

**Comment by gilbertohasnofb** [comment]
Spelling fixed.

@onox I am still having one problem which I can't spot the mistake. The menu toggle goes gray when the speed is above 5 knots (as expected), so you can't add chocks while in movement. But clicking on the wheel at any speed is adding the chocks! And I have such a clear conditional, see:

```
    <animation>
        <type>pick</type>
        <object-name>NoseWheel</object-name>
        <action>
            <button>0</button>
            <repeatable>false</repeatable>
            <condition>
                <and>
                    <not>
                        <property>sim/model/c172p/chock</property>
                    </not>
                    <less-than>
                        <property>velocities/groundspeed-kt</property>
                        <value>5.0</value>
                    </less-than>
                </and>
            </condition>
            <binding>
                <command>property-assign</command>
                <property>sim/model/c172p/chock</property>
                <value>true</value>
            </binding>
        </action>
```

Can you help me spot the mistake?

**Comment by onox** [comment]
Maybe you need to put `<expression>` around `<less-than>` block.

**Comment by Juanvvc** [comment]
I believe, but I'm not sure, that the condition block must be inside the binding block. Like in "bind this only if this condition is met"

**Comment by gilbertohasnofb** [comment]
No, it's not that.

**Comment by gilbertohasnofb** [comment]
> I believe, but I'm not sure, that the condition block must be inside the binding block. Like in "bind this only if this condition is met" 

Yep! That should be it!

**Comment by onox** [comment]
Hmm yeah, `<expression>` shouldn't be necessary for things like comparisons. So it's probably what @Juanvvc said.

**Comment by gilbertohasnofb** [comment]
@Juanvvc that solved it, thanks ludo!

File: Models/c172p.xml [hunk 1]
```diff
@@ -6587,5 +6587,273 @@
...
+    </animation>
+    <animation>
+        <type>pick</type>
+        <object-name>PitotTube</object-name>
+        <action>
+            <button>0</button>
+            <repeatable>false</repeatable>
+            <binding>
+                <condition>
+                    <and>
```
Comment (onox) [inline Models/c172p.xml]: You have this logic here multiple times (in different bindings), so better put it in a `<logic>` element then.

**Comment by gilbertohasnofb** [comment]
@onox one more thing, do you think I should add another condition for allowing the removal of chocks and pitot cover: not only ground speed below 5 knots, but also the wow conditions? I find it unrealistic that anyone will ever be in the air with ground speed below 5 knots and trying to play with chocks and pitot cover, but I could do that just to be able to sleep well :wink: 

And when I have a `<and>` or `<or>` block, can I have more than two expressions inside? That is, is the syntax below correct?:

```
<and>
    expression1
    expression2
    expression3
</and>
```

or do I need to do this?:

```
<and>
    expression1
    <and>
        expression2
        expression3
    </and>
</and>
```

File: Models/c172p.xml [hunk 2]
```diff
@@ -6587,5 +6587,273 @@
...
+                <label>Put Pitot Tube Cap</label>
+            </binding>
+        </hovered>
+    </animation>
+    
+    <!-- Wheel chock -->
+    <model>
+        <name>chock1</name>
+        <path>/Exterior/chock/chock.xml</path>
+        <condition>
```
Comment (onox) [inline Models/c172p.xml]: Tried removing the `<condition>`?

File: Models/c172p.xml (hunk 1 above)
Comment (gilbertohasnofb) [inline Models/c172p.xml]: Okay!

**Comment by onox** [comment]
You can put multiple things in `<and>` or `<or>`.

**Comment by gilbertohasnofb** [comment]
:+1:

File: Models/c172p.xml (hunk 2 above)
Comment (gilbertohasnofb) [inline Models/c172p.xml]: I will try

Comment (gilbertohasnofb) [inline Models/c172p.xml]: It works, just pushed a new commit.

File: Models/c172p.xml (hunk 1 above)
Comment (gilbertohasnofb) [inline Models/c172p.xml]: Done, see my new commit.

**Comment by onox** [comment]
@gilbertohasnofb Do you have anything to commit? I would like to merge `master` to `bug-303` to resolve the conflicts.

**Comment by gilbertohasnofb** [comment]
No, I am all done here, you can go ahead :smile:  The conflict will be easy to solve: I believe its related to the 3D baggages I added in a different branch and which we merged yesterday.

**Comment by onox** [comment]
You can pull :smile: Thanks for the work you have done @gilbertohasnofb

**Comment by gilbertohasnofb** [comment]
Thank YOU for all the patience with me :smile: I think I am starting to get the hang of it

**Comment by onox** [comment]
You want to keep the branch? Since #303 is still open.

**Comment by gilbertohasnofb** [comment]
Well, bug 303 still has other things to be done, so I'd say let's keep it. I will try to work on the wing straps (for securing the plane) when I have a chance, and I was thinking about working on that branch.

**Comment by onox** [comment]
Ok :+1:

**Comment by Juanvvc** [comment]
Thanks for all your work! This aircraft is becoming more and more amazing!

**Comment by gilbertohasnofb** [comment]
My pleasure!

**Comment by wkitty42** [comment]
what @Juanvvc said :smile_cat:

**Comment by gilbertohasnofb** [comment]
:smile:

**Pull request merged**